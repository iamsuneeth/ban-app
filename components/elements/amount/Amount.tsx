import React from "react";
import { View, Text, StyleSheet } from "react-native";
import NumberFormat from "react-number-format";

const CURRENCY_SYMBOL_MAP: { [k: string]: string } = {
  GBP: "£",
  INR: "₹",
  EUR: "€",
  USD: "$"
};

type Props = {
  amount: number;
  currency: string;
  size?: number;
};
export const Amount = ({ amount, currency, size = 16 }: Props) => {
  const amountText = amount.toFixed(2);
  const [integer, decimal] = amountText.split(".");

  return (
    <NumberFormat
      value={integer}
      displayType={"text"}
      thousandSeparator={true}
      prefix={CURRENCY_SYMBOL_MAP[currency]}
      renderText={value => (
        <View style={styles.container}>
          <Text style={{ fontSize: size }}>{value}</Text>
          <Text
            testID="decimal"
            style={{
              fontSize: Math.round(size * 0.8)
            }}
          >
            .{decimal}
          </Text>
        </View>
      )}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    flex: 1,
    alignItems: "flex-end"
  }
});
