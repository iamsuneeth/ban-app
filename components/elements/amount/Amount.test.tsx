import React from "react";
import { render } from "@testing-library/react-native";
import { Amount } from "./Amount";

describe("tests for Amount component", () => {
  test("should render correctly", () => {
    const { queryByTestId, baseElement } = render(
      <Amount amount={23} currency="GBP" />
    );
    expect(queryByTestId("decimal")).not.toBeNull();
    expect(baseElement).toMatchSnapshot();
  });
});
