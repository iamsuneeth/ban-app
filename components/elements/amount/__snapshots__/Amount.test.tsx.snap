// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Amount component should render correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Â£23
      </Text>
      <Text
        style={
          Object {
            "fontSize": 13,
          }
        }
        testID="decimal"
      >
        .
        00
      </Text>
    </View>
  </View>
</View>
`;
